### PR TITLE
Add Impute Transform

### PR DESCRIPTION
**Pull request by invokesus**
Will make separate PRs for 
- Decoupling from stack 
- Impute Transform docs

### CODE REVIEW

File: src/compile/data/impute.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import {DataFlowNode} from './dataflow';
+
+export class ImputeTransformNode extends DataFlowNode {
+  public clone() {
+    return new ImputeTransformNode(this.parent, duplicate(this.transform));
+  }
+
+  public producedFields() {
+    // typescript detects true as boolean type
+    return {[this.transform.impute]: true as true} ;
```
Comment (kanitw) [inline src/compile/data/impute.ts]: This casting is weird.  What's wrong with boolean type?

File: src/transform.ts [hunk 2]
```diff
@@ -212,6 +212,38 @@ export interface WindowTransform {
...
+   * __Default value:__:  `"value"`
+   */
+  method?: 'value' | 'mean' | 'median' | 'max' | 'min';
+  /**
+   * An optional array of fields by which to group the values.
+   * Imputation will then be performed on a per-group basis.
+   */
+  groupby?: string[];
+  /**
+   * The field value to use when the imputation method is `"value"`.
```
Comment (kanitw) [inline src/transform.ts]:
```
imputation `method`
```

File: src/compile/data/impute.ts (hunk 1 above)
Comment (invokesus) [reply src/compile/data/impute.ts]: Getting this error 
`Type '{ [x: string]: boolean; }' is not assignable to type 'Dict<true>'.`

**Comment by invokesus** [comment]
I use a GUI so doesn't take that much time .🙂

File: src/compile/data/impute.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/data/impute.ts]: Ok then.

**Comment by kanitw** [comment]
> I get lost if I get big merge conflicts so I like to make smaller commits.

Ok. That's fine.  Let's say you can either do an "atomic" PR (like this one--which mean I'm gonna ignore your commit messages and just merge squash) or a PR with one or a few atomic commits (which makes is much easier for reviewer).
  
If you do multiple atomic things in a PR, please make your PR contains truly atomic commits, instead subatomic commits as the latter is equivalent to having one big commits for a reviewer's perspective.

**Comment by kanitw** [comment]
Awesome :)

**Pull request merged**